c7n/resources/glue.py

```diff
@@ -189,6 +189,68 @@ class resource_type(TypeInfo):
     augment = universal_augment
 
 
+@GlueCrawler.action_registry.register('update-crawler')
+class UpdateGlueCrawler(BaseAction):
+    """Updates glue crawlers based on specified parameter
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: glue-crawler-update
+                resource: aws.glue-crawler
+                filters:
+                  - type: value
+                    key: CrawlerSecurityConfiguration
+                    value: null
+                actions:
+                  - type: update-crawler
+                    attributes:
+                        CrawlerSecurityConfiguration: glue-approved
+
+    """
+
+    schema = type_schema(
+        'update-crawler',
+        required=['attributes'],
+        attributes={
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'Role': {'type': 'string'},
+                'DatabaseName': {'type': 'string'},
+                'Schedule': {'type': 'string'},
+                'Configuration': {'type': 'string'},
+                'CrawlerSecurityConfiguration': {'type': 'string'},
+                'TablePrefix': {'type': 'string'},
+                'Classifier': {'type': 'array', 'items': {'type': 'string'}},
+                'SchemaChangePolicy': {
+                    'type': 'object',
+                    'additionalProperties': False,
+                    'properties': {
+                        'UpdateBehavior': {'type': 'string', 'enum': ['LOG', 'UPDATE_IN_DATABASE']},
+                        'DeleteBehavior': {
+                            'type': 'string', 'enum': [
+                                'LOG', 'UPDATE_IN_DATABASE', 'DEPRECATE_IN_DATABASE']}
+                    }
+                }
+            }
+        })
+
+    permissions = ('glue:UpdateCrawler',)
+    valid_origin_states = ('READY', 'FAILED')
+
+    def process(self, crawlers):
+        crawlers = self.filter_resources(crawlers, 'State', self.valid_origin_states)
+        client = local_session(self.manager.session_factory).client('glue')
+        for crawler in crawlers:
+            try:
+                client.update_crawler(Name=crawler['Name'], **self.data['attributes'])
```

Comment:
ideally we shouldn't be making an api call if there isn't any delta to the existing configuration

Reply:
Sounds good! I have modified the code to address this